spring-cloud-kubernetes-client-discovery/src/main/java/org/springframework/cloud/kubernetes/client/discovery/KubernetesDiscoveryClientUtils.java

```diff
@@ -61,4 +67,38 @@ static boolean matchesServiceLabels(V1Service service, KubernetesDiscoveryProper
 
 	}
 
+	/**
+	 * This adds the following metadata. <pre>
+	 *     - labels (if requested)
+	 *     - annotations (if requested)
+	 *     - metadata
+	 *     - service type
+	 * </pre>
+	 */
+	static Map<String, String> serviceMetadata(KubernetesDiscoveryProperties properties, V1Service service,
+			String serviceId) {
+
+		Map<String, String> serviceMetadata = new HashMap<>();
+		KubernetesDiscoveryProperties.Metadata metadataProps = properties.metadata();
+		if (metadataProps.addLabels()) {
+			Map<String, String> labelMetadata = keysWithPrefix(service.getMetadata().getLabels(),
+					metadataProps.labelsPrefix());
+			LOG.debug(() -> "Adding labels metadata: " + labelMetadata + " for serviceId: " + serviceId);
+			serviceMetadata.putAll(labelMetadata);
+		}
+		if (metadataProps.addAnnotations()) {
+			Map<String, String> annotationMetadata = keysWithPrefix(service.getMetadata().getAnnotations(),
+					metadataProps.annotationsPrefix());
+			LOG.debug(() -> "Adding annotations metadata: " + annotationMetadata + " for serviceId: " + serviceId);
+			serviceMetadata.putAll(annotationMetadata);
+		}
+
+		serviceMetadata.put(NAMESPACE_METADATA_KEY,
```

Comment:
metadata now includes two more fields, just like we do in the fabric8 implementation.